runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -3481,6 +3518,18 @@ J9::ValuePropagation::doDelayedTransformations()
 
    _valueTypesHelperCallsToBeFolded.deleteAll();
 
+   for (CallInfo* ci = _refinedMethodHandleINLMethodsToInline.getFirst(); ci; ci = ci->getNext())
+      {
+      TR_InlineCall newInlineCall(optimizer(), this);
+
+      if (!newInlineCall.inlineCall(ci->_tt, ci->_thisType, true, ci->_argInfo, 0))
```

Comment:
At warm and below this is ignoring two options that appear to be respected by the inlining of devirtualized calls: `TR_DisableInliningDuringVPAtWarm` and `getMaxSzForVPInliningWarm()`. It's not clear whether that's intentional. I think that for each of these options, either it should be respected here as well, or there should be a comment explaining why it isn't. Please let me know what you think

Reply:
I did not think that invokeBasic calls should be governed by the same size limits and options as devirtualized calls at warm and below, and therefore intentionally went with an unrestricted (well, subject to the restrictions in `inlineCall`) approach here. We can do the following if think we need to be able to restrict refined INL call inlining in VP:
* add the option `TR_DisableRefinedINLInliningDuringVPAtWarm` for INL methods instead of having `TR_DisableInliningDuringVPAtWarm` have an effect on such methods. Perhaps the latter could be renamed to `TR_DisableInliningDevirtualizedCallsDuringVPAtWarm`.
* set and use a higher max size for refined INL inlining in VP, with a new command line option to set it as well.